src/core/text/TextMetrics.js

```diff
@@ -255,6 +284,109 @@ export default class TextMetrics
         return width;
     }
 
+    /**
+     * trims breaking whitespaces from string
+     *
+     * @param  {string}  text  The text
+     * @return {string}  trimmed string
+     */
+    static trimRight(text)
+    {
+        for (let i = text.length - 1; i >= 0; i--)
+        {
+            const char = text[i];
+
+            if (!TextMetrics.isBreakingSpace(char))
+            {
+                break;
+            }
+
+            text = text.slice(0, -1);
+        }
+
+        return text;
+    }
+
+    /**
+     * Determines if char is a newline.
+     *
+     * @param  {string}  char  The character
+     * @return {boolean}  True if newline, False otherwise.
+     */
+    static isNewline(char)
+    {
+        const newlines = [
+            0x000A, // line feed
+            0x000D, // carriage return
+        ];
+
+        return (newlines.indexOf(char.charCodeAt(0)) >= 0);
+    }
+
+    /**
+     * Determines if char is a breaking whitespace.
+     *
+     * @param  {string}  char  The character
+     * @return {boolean}  True if whitespace, False otherwise.
+     */
+    static isBreakingSpace(char)
+    {
+        const breakingSpaces = [
+            0x0009, // character tabulation
+            0x0020, // space
+            0x2000, // en quad
+            0x2001, // em quad
+            0x2002, // en space
+            0x2003, // em space
+            0x2004, // three-per-em space
+            0x2005, // four-per-em space
+            0x2006, // six-per-em space
+            0x2008, // punctuation space
+            0x2009, // thin space
+            0x200A, // hair space
+            0x205F, // medium mathematical space
+            0x3000, // ideographic space
+        ];
+
+        return (breakingSpaces.indexOf(char.charCodeAt(0)) >= 0);
+    }
+
+    /**
+     * Splits a string into words, spaces and newLine characters
+     *
+     * @param  {string}  text       The text
+     * @return {array}  A tokenized array
+     */
+    static explodeIntoWordsSpacesAndNewlineChars(text)
```

Comment:
This name is pretty cumbersome, what about something simpler like "explode"

Reply:
ok shortened

Reply:
Done